server/src/main/java/org/elasticsearch/script/ScriptTermStats.java

```diff
@@ -73,8 +72,9 @@ public int matchedTermsCount() {
         int matchedTerms = 0;
 
         try {
+            advancePostings(docId);
             for (PostingsEnum postingsEnum : postingsSupplier.get()) {
-                if (postingsEnum != null && postingsEnum.advance(docId) == docId && postingsEnum.freq() > 0) {
+                if (postingsEnum != null && postingsEnum.docID() == docId && postingsEnum.freq() > 0) {
```

Comment:
How would it be possible for `postingsEnum.freq()` to return 0 for a match?

Reply:
This is right. Changed.